### DISCUSSION TITLE
Get RMSD between 2 known conformers

### DISCUSSION THREAD

**Discussion by TheophileGaudin** [rdkit/rdkit#5845] [Q&A] [answered]
Hi,

I have xyz coordinates extracted from other files that I can put into mol files and load in a MolFromMolFile function of rdkit. I would like to know whether I can compare two structures with the exact same atoms and bonds (in other words the exact same structure or SMILES) and get the RMSD. I need to cluster the structures that are identical or very similar (ie with low RMSD). I suspect that there is an easy way to do this with rdkit but I didn't find anything by searching on the web.

Does someone know how to carry out such a task?

**Comment by ptosco** [comment]
@TheophileGaudin The [`rdMolAlign`](http://rdkit.org/docs/source/rdkit.Chem.rdMolAlign.html?highlight=getbestrms#module-rdkit.Chem.rdMolAlign) module has several functions that can help you with that, e.g. [GetBestRMS()](http://rdkit.org/docs/source/rdkit.Chem.rdMolAlign.html?highlight=getbestrms#rdkit.Chem.rdMolAlign.GetBestRMS) and [CalcRMS()](http://rdkit.org/docs/source/rdkit.Chem.rdMolAlign.html?highlight=getbestrms#rdkit.Chem.rdMolAlign.CalcRMS).

**Reply by TheophileGaudin** [reply]
Thanks, but I couldn't find an example where someone actually uses this to load conformer into molecule object from external xyz coordinates and I can't deduce a proper way to use the functions from reading the documentation as it stands.

**Comment by ptosco** [comment, marked as answer]
@TheophileGaudin You can either clone an existing conformer or create a new one with the desired number of atoms and then replace coordinates with `Conformer.SetAtomPosition()`, e.g.:
```python
from rdkit import Chem
from rdkit.Geometry import Point3D

new_conformer = Chem.Conformer(existing_conformer)
# or new_conformer = Chem.Conformer(num_atoms)

# in this example xyz_coord_array is an array of (x, y, z) tuples
for i, (x, y, z) in enumerate(xyz_coord_array):
    new_conformer.SetAtomPosition(i, Point3D(x, y, z))
```

**Reply by TheophileGaudin** [reply]
Hi, thanks for the details. I ended up finding my reply in this github repository:
https://gist.github.com/tdudgeon/b061dc67f9d879905b50118408c30aac

In particular, I repurposed the cluster_conformers function for my own needs.
To enter conformers I took inspiration from some stackoverflow post that amounted to what you wrote just above.
